### ISSUE TITLE
Allow property access on immutable classes right after instantiation

### ISSUE THREAD

**Issue by BenMorel** [vimeo/psalm#4606]
I'm using immutable classes such as:

```php
/**
 * @psalm-immutable
 */
final class Table
{
    public string $schema;
    public string $name;
}
```

I use these classes as simple structs, and don't want to provide a constructor because some of them may have 10+ properties, which quickly becomes a mess, both the constructor and the calling side. Instead, I would do something more verbose:

```php
$table = new Table;
$table->schema = 'foo';
$table->name = 'bar';
```

Psalm complains about both `MissingConstructor` (which is alright, I've suppressed this one), bit also about `InaccessibleProperty`; I understand the rationale behind this, though **would it be possible to not report this when the property is set right after instantiation, in the same code block?**

If the property is set anywhere else, of course Psalm should report it. But when it's set right after `new`, it would be nice if Psalm could be silent. I cannot just suppress this error globally, as I do want Psalm to report all other usages.

https://psalm.dev/r/4a70c074d2

**Comment by andrew-demb** [comment]
See https://github.com/vimeo/psalm/issues/4348

**Comment by weirdan** [comment]
For PHP 8 you would just use promoted named properties: https://psalm.dev/r/0a29f1583b
